### PR TITLE
Fix registration of wildcard attrs

### PR DESCRIPTION
**Pull request by Cadair**
And fix the docs

### CODE REVIEW

**Review by nabobalis — APPROVED** [review]

**Comment by abhijeetmanhas** [comment]
Tests are failing because the loop needs `name` and `desc` to get values of registered attrs. If `desc` is empty, then it can't unpack.

**Comment by nabobalis** [comment]
Could we add a blank string?

File: sunpy/net/base_client.py [hunk 1]
```diff
@@ -128,6 +128,10 @@ def _print_client(client, html=False):
     t = Table(names=["Attr Type", "Name", "Description"],
               dtype=["U80", "U80", "U80"])
     for client_key in attrs.keys():
+        # Work around for * attrs having one length.
+        if len(attrs[client_key]) == 1 and attrs[client_key][0] == "*":
+            t.add_row((client_key.__name__, "All", "All valid attributes"))
+            continue
```
Comment (nabobalis) [inline sunpy/net/base_client.py:134]: A large hack.

Comment (nabobalis) [reply sunpy/net/base_client.py:134]: If somehow we got a '*' and a valid value, this hack would break.

**Review by nabobalis — CHANGES_REQUESTED** [review]
Needs a real review.

File: sunpy/net/base_client.py (hunk 1 above)
Comment (Cadair) [reply sunpy/net/base_client.py:134]: Why not move this into the loop below?
```python
for name, desc in attrs[client_key]:
    if name == "*":
        t.add_row((client_key.__name__, "All", "All valid attributes"))
        continue
    t.add_row((client_key.__name__, name, desc))
```
?

File: sunpy/net/base_client.py [hunk 2]
```diff
@@ -128,6 +128,10 @@ def _print_client(client, html=False):
     t = Table(names=["Attr Type", "Name", "Description"],
               dtype=["U80", "U80", "U80"])
     for client_key in attrs.keys():
+        # Work around for * attrs having one length.
+        if len(attrs[client_key]) == 1 and attrs[client_key][0] == "*":
+            t.add_row((client_key.__name__, "All", "All valid attributes"))
```
Comment (Cadair) [inline sunpy/net/base_client.py:133]: attributes feels weird?
```suggestion
            t.add_row((client_key.__name__, "All", "All valid values"))
```

File: sunpy/net/base_client.py (hunk 1 above)
Comment (nabobalis) [reply sunpy/net/base_client.py:134]: Do it.

Comment (nabobalis) [reply sunpy/net/base_client.py:134]: The issue with this was that if the length is one, there is no "desc" to loop over and thus errored.

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Online tests have failed with:
```python
FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunpy/net/jsoc/tests/test_jsoc.py::test_get_request - urllib.error.HTTPError: HTTP Error 500: Internal Server Error
FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunpy/net/jsoc/tests/test_jsoc.py::test_get_request_tar - urllib.error.HTTPError: HTTP Error 500: Internal Server Error
FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunpy/net/jsoc/tests/test_jsoc.py::test_check_request - urllib.error.HTTPError: HTTP Error 500: Internal Server Error
FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunpy/net/jsoc/tests/test_jsoc.py::test_results_filenames - urllib.error.HTTPError: HTTP Error 500: Internal Server Error
```

**Pull request merged**